tests/functional/intelligent-host-selection/01-periodic-clear-badhosts.t

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Test mainloop plugin periodically clears badhosts.
+# By setting the interval to << small we can also test whether job log retrieval
+# and remote tidy work.
+export REQUIRE_PLATFORM='loc:remote fs:indep'
+
+. "$(dirname "$0")/test_header"
+
+#-------------------------------------------------------------------------------
+set_test_number 6
+
+# We don't use the usual ``create_test_global_config`` because we need to pin
+# the result of ``get_random_platform_for_install_target(install_target)`` to
+# mixedhostplatform.
```

Comment:
Curious as to why this doesn't work for `create_test_global_config`

Reply:
If I remember rightly, `create_test_global_config` appends stuff to your test config (so it will over-write things if they are there). We need to have only one platform with `install target = ${CYLC_TEST_INSTALL_TARGET}`